src/cpu.c

```diff
@@ -619,3 +631,72 @@ void OPC_ADD_A_d8(void) {
 	ADD_A_n(immediate);
 	cpu.cycle_count += 8;
 }
+
+
+static void SUB_A_n(uint8_t n) {
+	uint8_t A = cpu.AF.bytes.high;
+	uint_fast16_t result = A - n;
```

Comment:
Probably we do not need `uint_fast16_t` here, do we? 🤔 
I think that `uint8_t` should do.

This is something that I will have to change for the `ADD_*` instruction in #15 too.
By doing this we implicitly rely on the (well-defined) wraparound behavior for unsigned integers of C. That should, to the best of my knowledge, mimic what the gameboy does.

Reply:
Exactly